tests/stages/test_discovery.py

```diff
@@ -56,12 +56,12 @@ class TestDiscovery:
     def _helper_find_projects_to_execute(
         self,
         files_touched: dict[str, str],
-        stage: Stage = TestStage.build(),
+        stage_name: str = "build",
```

Comment:
Small suggestion: maybe it's a good idea to keep these stage names defined somewhere in a test file to avoid magic strings all over the place? Something like `tests/utilities/test_stages.py`:
```python
build: string = "build"
test: string = "test"
deploy: string = "deploy"
```
```python
return find_projects_to_execute(
  …
  stage=test_stages.build,
  …
```

Or as an alternative, at least add parameter names to the method calls to make it easier to read:
```python
find_projects_to_execute(
  …
  stage="test",
  …
)
```

Reply:
Sorry, I actually put these stages back after going through the same chain of thought as you did here. But I forgot to change back these value changes